sdk/digitaltwins/azure-digitaltwins-core/src/samples/java/com/azure/digitaltwins/core/DigitalTwinsLifecycleAsyncSample.java

```diff
@@ -88,108 +100,220 @@ public static void main(String[] args) throws IOException, InterruptedException
+                    .doOnSuccess(aVoid -> System.out.println("Deleted digital twin: " + twinId))
+                    .doOnError(IgnoreNotFoundError)
+                    .doOnTerminate(deleteTwinsLatch::countDown)
+                    .subscribe();
+
+                // Wait until the latch has been counted down for each async delete operation, signifying that the async call has completed successfully.
+                deleteTwinsLatch.await(MaxWaitTimeAsyncOperationsInSeconds, TimeUnit.SECONDS);
+            }
+        }
+    }
+
+    /**
+     * Delete models created by FullLifecycleSample for the ADT service instance.
+     * @throws InterruptedException If the current thread is interrupted while waiting to acquire latch.
+     */
+    public static void deleteAllModels() throws InterruptedException {
+        System.out.println("DELETING MODELS");
+
+        // This is to ensure models are deleted in an order such that no other models are referencing it.
+        List<String> models = asList(RoomModelId, WifiModelId, BuildingModelId, FloorModelId, HvacModelId);
 
+        // Call APIs to delete the models.
+        // Note that we are blocking the async API call. This is to ensure models are deleted in an order such that no other models are referencing it.
+        models
+            .forEach(modelId -> {
                 try {
-                    // Verify that the list relationships and list incoming relationships async operations have completed.
-                    if (deleteRelationshipsSemaphore.tryAcquire(2, MaxWaitTimeAsyncOperationsInSeconds, TimeUnit.SECONDS)) {
-                        // Now the digital twin should be safe to delete
-
-                        // Call APIs to delete the twins.
-                        client.deleteDigitalTwin(twinId)
-                            .doOnSuccess(aVoid -> {
-                                System.out.println("Deleted digital twin: " + twinId);
-                                deleteTwinsSemaphore.release();
-                            })
-                            .doOnError(throwable -> {
-                                if (throwable instanceof ErrorResponseException && ((ErrorResponseException) throwable).getResponse().getStatusCode() == HttpStatus.SC_NOT_FOUND) {
-                                    deleteTwinsSemaphore.release();
-                                } else {
-                                    System.err.println("Could not delete digital twin " + twinId + " due to " + throwable);
-                                }
-                            })
-                            .subscribe();
+                    client.deleteModel(modelId).block();
+                    System.out.println("Deleted model: " + modelId);
+                } catch (ErrorResponseException ex) {
+                    if (ex.getResponse().getStatusCode() != HttpStatus.SC_NOT_FOUND) {
+                        System.err.println("Could not delete model " + modelId + " due to " + ex);
```

Comment:
If Java doesn't have string interpolation, I'd think this would be much more readable as a string format (like below).